charts/partials/templates/_network-validator.tpl

```diff
@@ -0,0 +1,24 @@
+{{- define "partials.network-validator" -}}
+name: linkerd-network-validator
+image: {{.Values.proxy.image.name}}:{{.Values.proxy.image.version | default .Values.linkerdVersion }}
+imagePullPolicy: {{.Values.proxy.image.pullPolicy | default .Values.imagePullPolicy}}
+securityContext:
+  runAsUser: 2104
```

Comment:
Don't we accept a proxy configuration to control this UID? We probably need to use the same configuration?

https://github.com/linkerd/linkerd2/blob/9657d867650cc58e34307355239bcb3f19912579/charts/linkerd2-cni/values.yaml#L21-L22

I'm surprised this works with 2104... Does it? Why?

Reply:
The validator runs with these settings, I wasn't able to verify it ran with that exact user id. Let me change it to wire in proxyUID from values and see if I can get it to tell me the user id before the container terminates.

Reply:
So I've verified that proxy.uid is wired through to the yaml config but I also tested in AKS with  `runAsUser: 9999` (a random high number) and validator was scheduled and ran without complaint.

Reply:
I think it needs to be `.Values.proxyUID` -- for whatever reason this setting isn't under `proxy: ...`

> I also tested in AKS with runAsUser: 9999 (a random high number) and validator was scheduled and ran without complaint.

OOOOH, you know what? I'm wrong about everything. We explicitly **don't** want to run this as the proxy UID, since that will bypass the iptables rules.

Sorry for the bad suggestion.  Let's use 65534, which is `nobody`, where we don't actually care about the UID.

Reply:
{{ .Values.proxy.uid }} worked for me but I've changed it to 65534

Reply:
Blegh, you're right. I was looking at the CNI values.yaml file :(